### ISSUE TITLE
Add SlicerAPI / Build System to documentation

### ISSUE THREAD

**Issue by jcfr** [Slicer/Slicer#5198]
Instead of adding documentation like I did here:
https://github.com/Slicer/Slicer/pull/5194/commits/713c2ac73ba61c93e2275f32d18fbef726d10e03

I am thinking to instead to something like this:

```
# CMake variables supported by this external project:
#
# * LZMA_INCLUDE_DIR
# * LZMA_LIBRARY
# 
# To learn more about external project configuration, see https://slicer.readthedocs.io/en/latest/developer_guide/api.html#build-system
```

We would basically add a new "Build System" category to the SlicerAPI with a sub-section called external project where we would explain how the external project generally work.

_Originally posted by @jcfr in https://github.com/Slicer/Slicer/pull/5194#issuecomment-697008292_

**Comment by lassoan** [comment]
Good idea. Maybe we could use the existing https://slicer.readthedocs.io/en/latest/developer_guide/build_instructions/overview.html page for now and split into more pages as the content grows too long for a single page.
